pkg/build/controller/image_change_controller_test.go

```diff
@@ -423,3 +423,12 @@ func mockImageChangeController(buildcfg *buildapi.BuildConfig, imageStream *imag
 		BuildConfigInstantiator: mockBuildConfigInstantiator(buildcfg, imageStream, image),
 	}
 }
+
+func TestGetImageStreamNameFromReference(t *testing.T) {
+	ref := &kapi.ObjectReference{}
+	ref.Name = "testImage@id:/registry.com/namespace/imagename"
```

Comment:
Could you please add more valid cases like `ruby-20-centos7@sha256:603bfa418e1ff0a486f3d795a8b69a2f8f5eed488fbf5c286bffd8ab77b8e7a5`, `ruby-20-centos7`, `repo@id`, `is:tag`?